### PR TITLE
Implement Diff Fetching Functionality and Frontend Display

### PR DESCRIPTION
**Pull request by kmesiab**
## Initial Screen

image

## Fetched Pull Request View

image

## Overview

This pull request introduces key features to the Orion project, tackling issues [#13](https://github.com/kmesiab/orion/issues/13), [#12](https://github.com/kmesiab/orion/issues/12), and [#10](https://github.com/kmesiab/orion/issues/10). The new functionality spans both backend and frontend components, enabling efficient fetching and display of GitHub pull request diffs.

### Backend Diff Fetching Function

- **Description**: Implemented a Go function exposed to TypeScript via Wails binding. This function fetches Git diffs from a given GitHub PR URL and generates a diff view, laying the groundwork for frontend interactions.
- **Issue Link**: [#13](https://github.com/kmesiab/orion/issues/13)

### Frontend Input and Trigger Button

- **Description**: Added a text input field and a button on the frontend. These components allow users to enter a GitHub PR URL and trigger the backend process for diff fetching.
- **Issue Link**: [#12](https://github.com/kmesiab/orion/issues/12)

### Styled Diff Viewer

- **Description**: Developed a user-friendly view in React to display Git diffs. The viewer, styled with SCSS, leverages `react-diff-view` for a clear and organized presentation.
- **Issue Link**: [#10](https://github.com/kmesiab/orion/issues/10)

## Technical Details

- The backend Go function integrates seamlessly with the GitHub API for PR diff retrieval.
- Frontend elements are crafted using React, with state management via React Hooks.
- The diff viewer component employs SCSS for a consistent and aesthetically pleasing display.

## Testing and Validation

- Conducted manual tests to ensure functionality and integration.
- Visually validated the diff viewer for clarity and effectiveness.

This PR sets a solid foundation for future enhancements, augmenting Orion's capabilities.

---

Feel free to adjust or enhance this description as needed to best fit your project's needs.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new pull request processing feature in the application interface.
	- Added a `DiffViewer` component to visually display text differences.

- **Improvements**
	- Enhanced the build process to include necessary dependency installations.
	- Updated application styling for better visual appeal and responsiveness.

- **Bug Fixes**
	- Adjusted application initialization to ensure proper service setup.

- **Documentation**
	- Updated the hash in `package.json.md5` to reflect changes in dependencies.

- **Refactor**
	- Renamed and changed the functionality of several application methods to align with the new pull request feature.
	- Modified GitHub Actions workflows to improve build and test processes.

### CODE REVIEW

File: github/github.go
Comment (kmesiab) [reply github/github.go:39]: Good catch.  Fixed and amended the commit.

File: github/github_test.go
Comment (kmesiab) [reply github/github_test.go:66]: I will address testing after we figure out what we're doing. :)

File: app.go
Comment (kmesiab) [reply app.go]: This has been fixed and submitted in an amended commit.

File: main_test.go
Comment (kmesiab) [reply main_test.go:1]: Yes, please do submit an issue for this.  Thank you.

File: github/github_test.go
Comment (kmesiab) [reply github/github_test.go:66]: Please submit an issue for this.  Thank you.

**Comment by kmesiab** [comment]
@frankie-mur This PR is open for discussion.

**Review by frankie-mur — DISMISSED** [review]
LGTM! Great job @kmesiab, Comments are great, and wow thats alot of css 😆 did you write all of that yourself? Should we consider using some type of css lib? 
Also noticed you have made the github diff parsing logic into its own lib!

File: frontend/src/App.jsx
Comment (frankie-mur) [inline frontend/src/App.jsx:8]: setUrl?

**Comment by kmesiab** [comment]
> LGTM! Great job @kmesiab, Comments are great, and wow thats alot of css 😆 did you write all of that yourself? Should we consider using some type of css lib? Also noticed you have made the github diff parsing logic into its own lib!

We absolutely MUST use a library 😆😆😆.  I have cleaned up the CSS in another commit i will merge with this one shortly.  It adds tabs.

Definitely not my strong suit ;)

File: frontend/src/App.jsx
Comment (kmesiab) [reply frontend/src/App.jsx:8]: Oops, yes it should.  Good catch.

Comment (kmesiab) [reply frontend/src/App.jsx]: Fixed and squashed.  I've also updated the screenshot after merging in the PR that adds tabs.

File: frontend/src/style.scss
Comment (kmesiab) [reply frontend/src/style.scss]: Removed and squashed.

File: frontend/src/style.scss
Comment (kmesiab) [reply frontend/src/style.scss:135]: Make a ticket for this

**Comment by kmesiab** [comment]
@frankie-mur updates ready for review.  Hit the approve button when you're ready.

**Review by frankie-mur — APPROVED** [review]

**Pull request merged**
